app/models/ems_refresh.rb

```diff
@@ -173,7 +173,13 @@ def self.queue_merge(targets, ems, create_task = false)
     # Items will be naturally serialized since there is a dedicated worker.
     MiqQueue.put_or_update(queue_options) do |msg, item|
       targets = msg.nil? ? targets : msg.data.concat(targets)
-      targets.uniq! if targets.size > 1_000
+      if targets.size > 1_000
+        manager_refresh_targets, application_record_targets = targets.partition { |key, _| key == "ManagerRefresh::Target" }
+        application_record_targets.uniq!
+        manager_refresh_targets.uniq! { |_, value| value.values_at(:manager_id, :association, :manager_ref) }
+
+        targets = application_record_targets + manager_refresh_targets
+      end
```

Comment:
Can you move this to a private helper method?  queue_merge is long enough already :/